### PR TITLE
Rox 25519: Part-3: Prune orphaned node CVEs

### PR DESCRIPTION
**Pull request by charmik-redhat**
### Description

This is the part-2 of a 3 part change to prevent immediate removal of orphaned Node CVEs.

Part-1 [#12283]: Added proto changes, generated store and datastore. Added environment variables to enable/disable these changes.

Part-2 [#12372]: 
- Updates node store implementation to mark orphaned CVEs and restore them when rediscovered.
- Updates GraphQL implementations to exclude orphaned CVEs

Part-3: Adds pruning job to periodically prune orphaned Node CVEs that have not been restored for a set duration. 

### User-facing documentation

- [X] CHANGELOG update is not needed
- [X] documentation is not needed

### Testing and quality

- [X] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [X] CI results are inspected

#### Automated testing

- [X] added unit tests
- [X] modified existing tests

#### How I validated my change

Unit tests and manual testing

### CODE REVIEW

File: central/cve/node/datastore/datastore_impl.go [hunk 1]
```diff
@@ -110,6 +113,36 @@ func (ds *datastoreImpl) GetBatch(ctx context.Context, ids []string) ([]*storage
 	return cves, nil
 }
 
+func (ds *datastoreImpl) UpsertMany(ctx context.Context, cves []*storage.NodeCVE) error {
```
Comment (dashrews78) [inline central/cve/node/datastore/datastore_impl.go:116]: I know it isn't super exciting, but should probably add this to the unit tests.

File: central/pruning/pruning.go [hunk 2]
```diff
@@ -1031,6 +1038,32 @@ func (g *garbageCollectorImpl) pruneLogImbues() {
 	postgres.PruneLogImbues(pruningCtx, g.postgres, logImbueWindow)
 }
 
+func (g *garbageCollectorImpl) pruneOrphanedNodeCVEs() {
+	retentionDays := int64(env.OrphanedCVEsRetentionDurationDays.IntegerSetting())
+	if retentionDays < 0 {
+		log.Errorf("Invalid value of ROX_ORPHANED_CVES_RETENTION_DURATION_DAYS env var: %d, value should be >= 0", retentionDays)
```
Comment (dashrews78) [inline central/pruning/pruning.go:1044]: Warning maybe?  Should we return?  Should we use a default instead in this case?

Comment (charmik-redhat) [reply central/pruning/pruning.go:1044]: Good catch. I did not return or do anything with it after the error. I think we can fallback to default value in that case.

File: central/pruning/pruning.go [hunk 3]
```diff
@@ -1031,6 +1038,34 @@ func (g *garbageCollectorImpl) pruneLogImbues() {
...
+	if retentionDays < 0 {
+		log.Warnf("Invalid value of ROX_ORPHANED_CVES_RETENTION_DURATION_DAYS env var: %d, value should be >= 0. "+
+			"Using default value %d days", retentionDays, env.OrphanedCVEsRetentionDurationDays.DefaultValue())
+		retentionDays = int64(env.OrphanedCVEsRetentionDurationDays.DefaultValue())
+	}
+
+	query := search.NewQueryBuilder().AddBools(search.CVEOrphaned, true).AddDays(search.CVEOrphanedTime, retentionDays).ProtoQuery()
+	results, err := g.nodeCVEStore.Search(pruningCtx, query)
+	if err != nil {
+		log.Error(errors.Wrap(err, "Pruning orphaned node CVEs"))
```
Comment (dashrews78) [inline central/pruning/pruning.go:1052]: nit:  should probably return here.

**Review by dashrews78 — APPROVED** [review]
one small nit.  Looks good.

**Pull request merged**